mrp_production_force_bom/models/mrp_production.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def _check_unexpected_products(
+        self, move_raw_product_ids, bom_product_ids, error_messages
+    ):
+        unexpected_products = move_raw_product_ids - bom_product_ids
+        if unexpected_products:
+            error_messages.append(
+                _("For order '%s': Some components are not part of the BOM.")
+                % self.name
+            )
+        return error_messages
+
+    def _check_missing_products(
+        self, move_raw_product_ids, bom_product_ids, error_messages
+    ):
+        missing_products = bom_product_ids - move_raw_product_ids
+        if missing_products:
+            error_messages.append(
+                _(
+                    "For order '%s': Some components from the BOM are missing"
+                    " in the raw materials."
+                )
+                % self.name
+            )
+        return error_messages
+
+    def _force_bom(self):
+        param = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("mrp_production_force_bom.force_bom")
+        )
+        if param == "1":
+            return True
+        return False
+
+    def action_confirm(self):
+        if not self._force_bom():
+            return super().action_confirm()
+        error_messages = []
+        for production in self:
```

Comment:
```suggestion
        for production in self:
            if production._skip_bom_check():
                continue
```
Can we do this to provide a hook point.